test/widgets/settings_test.dart

```diff
@@ -24,6 +24,30 @@ void main() {
     await tester.pump();
   }
 
+  group('BrowserPreference', () {
+    Finder useExternalBrowserSwitchFinder = find.ancestor(
+      of: find.text('Use external browser'),
+      matching: find.byType(SwitchListTile));
+
+    testWidgets('smoke', (tester) async {
```

Comment:
Let's also test that the switch is in the right place, and responds correctly, when the setting hasn't been set yet (the part meant by "effective" in `effectiveBrowserPreference`)